CodingTracker.kkvzx/Presenters/CodingSessionPresenter.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CodingTracker.Data;
+using CodingTracker.model;
+using CodingTracker.Views;
+
+namespace CodingTracker.Presenters;
+
+public class CodingSessionPresenter(CodingSessionsView view, CodingTrackerRepository repository)
+{
+    private DateTime? CurrentSessionStart { get; set; }
+
+    public void Run()
+    {
+        repository.CreateTable();
+
+        while (true)
+        {
+            int sessionCount = repository.GetRecordsCount();
+            string selectedOption = view.ShowMenu(IsSessionInProgress, sessionCount);
+
+            switch (selectedOption)
+            {
+                case "Start new session": HandleStartNewSession(); break;
+                case "Stop running session": HandleStopSession(); break;
+                case "Add session": HandleAddCodingSessionSelect(); break;
+                case "Show all sessions": HandleShowAllSessionsSelect(); break;
+                case "Update session": HandleUpdateSessionSelect(); break;
+                case "Delete session": HandleDeleteSession(); break;
+                case "Show report": HandleShowReport(); break;
+                case "Exit": return;
+                default: HandleInvalidChoice(); break;
+            }
```

Comment:
Then with `MenuOptions` enum it would look like this
```suggestion
            switch (selectedOption)
            {
                case MenuOptions.StartSession: HandleStartNewSession(); break;
                case MenuOptions.StopSession: HandleStopSession(); break;
                case MenuOptions.RunSession: HandleAddCodingSessionSelect(); break;
                case MenuOptions.ShowAllSessions: HandleShowAllSessionsSelect(); break;
                case MenuOptions.UpdateSession: HandleUpdateSessionSelect(); break;
                case MenuOptions.DeleteSession: HandleDeleteSession(); break;
                case MenuOptions.ShowReport: HandleShowReport(); break;
                case MenuOptions.Exit: return;
                default: HandleInvalidChoice(); break;
            }
```